runtime/lua/vim/filetype/detect.lua

```diff
@@ -1322,6 +1322,28 @@ function M.txt(bufnr)
   end
 end
 
+function M.typ(bufnr)
+  if vim.g.filetype_typ then
+    return vim.g.filetype_typ
+  end
+
+  -- Look for SQL type definition syntax
+  for _, line in ipairs(getlines(bufnr, 1, 200)) do
+
+    -- SQL type files may define the casing
+    if line:find('^CASE%s%==%s%=%(SAME%|LOWER%|UPPER%|OPPOSITE%)$') then
```

Comment:
You can combine this with `findany`.

Reply:
(and no need for the comments; they're not really adding anything that isn't obvious from the code)

Reply:
Is it ok for you like this ?